src/components/ProjectSearch/index.js

```diff
@@ -20,84 +23,256 @@ import {
+        <span className="file-path">{getRelativePath(file.filepath)}</span>
+        <span className="matches-summary">{matches}</span>
+      </div>
+    );
   }
 
-  isProjectSearchEnabled() {
-    return this.props.activeSearch === "project";
+  renderMatch(match: Match, focused: boolean) {
+    if (focused) {
+      this.focusedItem = { match };
+    }
+    return (
+      <div
+        className={classnames("result", { focused })}
+        onClick={() => setTimeout(() => this.selectMatchItem(match), 50)}
+      >
+        <span className="line-number" key={match.line}>
+          {match.line}
+        </span>
+        {this.renderMatchValue(match)}
+      </div>
+    );
   }
 
-  renderTextSearch() {
-    const {
-      sources,
-      results,
-      status,
-      searchSources,
-      closeProjectSearch,
-      selectLocation,
-      textSearchQuery
-    } = this.props;
+  renderMatchValue(lineMatch: Match) {
+    return highlightMatches(lineMatch);
+  }
+
+  renderResults() {
+    const results = this.getResults().filter(
+      result => result.matches.length > 0
+    );
+
+    const { status } = this.props;
+
+    function getFilePath(item: Match, index?: number) {
+      return item.filepath
+        ? `${item.sourceId}-${index || "$"}`
+        : `${item.sourceId}-${item.line}-${item.column}-${index || "$"}`;
+    }
+
+    const renderItem = (item, depth, focused, _, expanded, { setExpanded }) => {
+      return item.filepath
+        ? this.renderFile(item, focused, expanded, setExpanded)
+        : this.renderMatch(item, focused);
+    };
+    if (results.length && status === statusType.done) {
+      return (
+        <ManagedTree
+          getRoots={() => results}
+          getChildren={file => file.matches || []}
+          itemHeight={24}
+          autoExpandAll={true}
+          autoExpandDepth={1}
+          getParent={item => null}
+          getPath={getFilePath}
+          renderItem={renderItem}
+        />
+      );
+    } else if (
+      (this.props.query && !results.length) ||
+      status === statusType.fetching
+    ) {
+      return (
+        <div className="no-result-msg absolute-center">
+          {L10N.getStr("projectTextSearch.noResults")}
+        </div>
+      );
+    }
+  }
+
+  renderInput() {
+    const resultCount = this.getResultCount();
 
     return (
-      <TextSearch
-        sources={sources}
-        results={results.toJS()}
-        status={status}
-        searchSources={searchSources}
-        closeProjectSearch={closeProjectSearch}
-        selectLocation={selectLocation}
-        query={textSearchQuery}
+      <SearchInput
+        query={this.state.inputValue}
+        count={resultCount}
+        placeholder={L10N.getStr("projectTextSearch.placeholder")}
+        size="big"
+        summaryMsg={
+          this.props.query !== ""
+            ? L10N.getFormatStr("sourceSearch.resultsSummary1", resultCount)
+            : ""
+        }
+        onChange={e => this.inputOnChange(e)}
```

Comment:
could we do? `onChange={this.inputOnChange}`

Reply:
Yeah I think so. Like listed above, lots was moved without a ton of scrutiny on my part of whether it's done the best way.